src/main/java/org/reaktivity/nukleus/http2/internal/stream/Http2ServerFactory.java

```diff
@@ -2190,13 +2248,15 @@ private void onDecodeData(
                 }
                 else
                 {
-                    if (dataLength > 0)
+                    if (payloadLength > 0)
                     {
-                        final DirectBuffer payload = http2Data.payload();
-                        exchange.doRequestData(traceId, authorization, payload, 0, payload.capacity());
+                        final int dataSent = exchange.doRequestData(traceId, authorization, payload, 0, payloadLength);
```

Comment:
Suggest using a `MutableInteger` for `payloadLength`, initialized to `payload.capacity()`, but filled in with exactly how much was sent by `doRequestData`.
That seems better than using the return value from `exchange.doRequestData(...)` as we really are giving a candidate payload length and then having the maximum sent be adjusted due to flow control limits.